From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Fedora Python maintainers <python-devel@lists.fedoraproject.org>
Date: Wed, 15 Jul 2020 15:18:01 +0200
Subject: [PATCH] python-2.6.4-distutils-rpath.patch

00017 #
Fixup distutils/unixccompiler.py to remove standard library path from rpath:
Adapted from Patch0 in ivazquez' python3000 specfile, removing usage of
super() as it's an old-style class
---
 Lib/distutils/unixccompiler.py | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/Lib/distutils/unixccompiler.py b/Lib/distutils/unixccompiler.py
index 3af540e5813..1c9cfe89a21 100644
--- a/Lib/distutils/unixccompiler.py
+++ b/Lib/distutils/unixccompiler.py
@@ -85,6 +85,16 @@ class UnixCCompiler(CCompiler):
     if sys.platform == "cygwin":
         exe_extension = ".exe"
 
+    def _fix_lib_args(self, libraries, library_dirs, runtime_library_dirs):
+        """Remove standard library path from rpath"""
+        libraries, library_dirs, runtime_library_dirs = \
+            CCompiler._fix_lib_args(self, libraries, library_dirs,
+                                    runtime_library_dirs)
+        libdir = sysconfig.get_config_var('LIBDIR')
+        if runtime_library_dirs and (libdir in runtime_library_dirs):
+            runtime_library_dirs.remove(libdir)
+        return libraries, library_dirs, runtime_library_dirs
+
     def preprocess(self, source,
                    output_file=None, macros=None, include_dirs=None,
                    extra_preargs=None, extra_postargs=None):
